### ISSUE TITLE
GIF image disappear when set url image as button background 

### ISSUE THREAD

**Issue by newpgt** [rs/SDWebImage#1732]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / tvos
 Platform Version        | e.g. 8.0
 SDWebImage Version      | e.g. 3.7
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 8.1
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

method:
buttonview.sd_setBackgroundImageWithURL(NSURL(string: imageString), forState: UIControlState.Normal, placeholderImage: PlaceholderImage)

The uibutton is set as element in uitableviewcell. When the url image is normal every thing works fine. But when the url image format is GIF, the image only appear once and disappear by either click the button or scroll tableview.

**Comment by bpoplauschi** [comment]
Please retry with 4.0.0, we have replaced our GIF component with FLAnimatedImage.

**Comment by ikesyo** [comment]
I'm facing the same or similar issue with the latest version (4.2.3).
